packages/generator/src/complex-type/namespace.ts

```diff
@@ -33,7 +33,21 @@ function factoryFunction(
 }
 
 function getJsonType(complexType: VdmComplexType): string {
-  return '{ [keys: string]: FieldType }';
+  const unionOfAllPropertyTypes = complexType.properties
+    .map(prop => prop.jsType)
+    .reduce(
+      (uniqueList: string[], item) =>
+        uniqueList.includes(item) ? uniqueList : [...uniqueList, item],
+      ['undefined', 'null']
+    )
+    .sort()
+    .reduce(
+      (typeList, item) =>
+        typeList.length === 0 ? item : typeList + ' | ' + item,
+      ''
+    );
+
+  return `{ [keys: string]: ${unionOfAllPropertyTypes} }`;
```

Comment:
As mentioned above I think it would be simpler to just have FieldType | NestedTypes (if necessary). I guess you can somehow find out what types are not in FieldType and maybe combine it with my suggestion above?